src/firestore/observable/fromRef.ts

```diff
@@ -1,25 +1,26 @@
 import { Observable, Subscriber } from 'rxjs';
 import { DocumentReference, Query, Action, Reference, DocumentSnapshot, QuerySnapshot } from '../interfaces';
 import { map, share } from 'rxjs/operators';
+import { firestore } from 'firebase';
 
-function _fromRef<T, R>(ref: Reference<T>): Observable<R> {
+function _fromRef<T, R>(ref: Query, options?: firestore.SnapshotListenOptions): Observable<R> {
   return new Observable(subscriber => {
-    const unsubscribe = ref.onSnapshot(subscriber);
+    const unsubscribe = ref.onSnapshot(options || {}, subscriber as any)
     return { unsubscribe };
   });
 }
 
-export function fromRef<R>(ref: DocumentReference | Query) {
-  return _fromRef<typeof ref, R>(ref).pipe(share());
+export function fromRef<R>(ref: any, options?: firestore.SnapshotListenOptions) {
+  return _fromRef<typeof ref, R>(ref, options).pipe(share());
 }
 
-export function fromDocRef<T>(ref: DocumentReference): Observable<Action<DocumentSnapshot<T>>>{
-  return fromRef<DocumentSnapshot<T>>(ref)
-    .pipe(
-      map(payload => ({ payload, type: 'value' }))
-    );
+export function fromDocRef<T>(ref: DocumentReference): Observable<DocumentSnapshot<T>>{
```

Comment:
Do we want to drop the `payload`/`type` in this commit? If so let's make sure to address the docs.